core/client/fs/src/main/java/alluxio/client/file/BaseFileSystem.java

```diff
@@ -394,18 +394,18 @@ public FileInStream openFile(AlluxioURI path)
   public FileInStream openFile(AlluxioURI path, OpenFilePOptions options)
       throws FileDoesNotExistException, IOException, AlluxioException {
     checkUri(path);
-    URIStatus status = getStatus(path);
-    if (status.isFolder()) {
-      throw new FileDoesNotExistException(
-          ExceptionMessage.CANNOT_READ_DIRECTORY.getMessage(status.getName()));
-    }
-    // getStatus should have loaded path level configs.
-    OpenFilePOptions mergedOptions = FileSystemOptions.openFileDefaults(
-        mFsContext.getPathConf(path)).toBuilder().mergeFrom(options).build();
-    InStreamOptions inStreamOptions = new InStreamOptions(status, mergedOptions,
-        mFsContext.getPathConf(path));
-    // FileSystemContext reinitialization is blocked during construction and close of FileInStream.
-    return new FileInStream(status, inStreamOptions, mFsContext);
+    return rpc(client -> {
+      URIStatus status = client.getStatus(path, GetStatusPOptions.getDefaultInstance());
```

Comment:
Nop, I changed to use path config here. My intention is to use the `rpc` convention to ensure the mFsContext used in `openFile` is consistent, otherwise, there is a possibility that `getStatus` uses an old version of `mFsContext`, then `mFsContext` is reinitialized, then the `FileInStream` uses the new version.